rebasehelper/specfile.py

```diff
@@ -403,15 +404,79 @@ def write_updated_patches(self, patches):
                 else:
                     patch = None
                 if patch:
-                    fields[1] = os.path.join(settings.REBASE_HELPER_RESULTS_DIR, patch_name)
+                    fields[1] = os.path.join(settings.REBASE_HELPER_NEW_SOURCES_DIR, patch_name)
                     self.spec_content[index] = ' '.join(fields) + '\n'
-                    modified_patches.append(patch_num)
-
+                    self.modified_patches.append(patch_num)
         self._comment_out_patches(removed_patches)
-        self._correct_rebased_patches(modified_patches)
+        self._correct_rebased_patches(self.modified_patches)
         #  save changes
         self.save()
 
+    @staticmethod
+    def filter_modified_patches(patches, spec_file):
+        """
+        Removes modified patches from *patches* which are already ready in the directory
+
+        :param patches: all patches not commented in the rebase_spec_file
+        :param spec_file: spec_file object
+        :return: List of unmodified used patches
+        """
+        copy_patches = []
+
+        for patch in patches:
+            copy = True
+            for pnum in spec_file.modified_patches:
```

Comment:
`spec_file.modified_patches` will always be empty, since `write_updated_patches()` method is called only on `rebase_spec_file`.